### ISSUE TITLE
Sqlness test with remote WAL fails

### ISSUE THREAD

**Issue by killme2008** [GreptimeTeam/greptimedb#4304]
### What type of bug is this?

Unexpected error

### What subsystems are affected?

Distributed Cluster

### Minimal reproduce step

https://github.com/GreptimeTeam/greptimedb/actions/runs/9814869931/job/27104613264?pr=4303

There is an error in `global-bg-worker` runtime:

```
thread 'global-bg-worker' has overflowed its stack
fatal runtime error: stack overflow
```

### What did you expect to see?

Run successfully

### What did you see instead?

Error

### What operating system did you use?

Not relevant

### What version of GreptimeDB did you use?

Not relevant

**Comment by killme2008** [comment]
Looks weird, only failed with remote WAL. I run it locally, and all is good.

**Comment by killme2008** [comment]
After reverting `lz4` compression, it works

https://github.com/GreptimeTeam/greptimedb/pull/4303/commits/180dda13faad3713d2597ab9d93235a90ecb9dcd

Looks like it's introduced by #4294

**Comment by WenyXu** [comment]
Another failure: https://github.com/GreptimeTeam/greptimedb/actions/runs/9833718316/job/27144618061

**Comment by WenyXu** [comment]
fixed in #4317
